src/shogun/lib/StoppableSGObject.cpp

```diff
@@ -60,6 +68,7 @@ void CStoppableSGObject::on_next()
 void CStoppableSGObject::on_pause()
 {
 	m_pause_computation_flag.store(true);
+	this->serialize_machine(std::string(this->get_name()) + ".txt");
```

Comment:
This is good but in my opinion, the serialization should be placed into ``on_pause_impl()`` instead :) devs may want to provide other behaviours for algorithm instead of serializing them by default (and to do it now they would have to override the ``on_pause`` method, which is no good).

Reply:
i second what @geektoni says above.... why do you want to automatically serialize a machine and why that name and why in a format of ascii and why extension of .txt and what happens if the file already exists...

Reply:
second @geektoni comment

Also agree with @vigsterkr The ascii format should not be compulsory, as discussed in another thread. Same goes for filename. Also some error handling will be needed

Reply:
@vigsterkr the idea is to serialize in Ascii as default when pausing but the user can always change that, infact even specify what type he wants before pausing :) I am still working on it atm

Reply:
mmm sorry but if i want to pause my training i dont want an object to touch my FS and do any sort of IO without my explicit request. i dont understand why a pause should trigger IO and serialization of the object itself. if one pauses a procedure, then it just pauses the procedure, having these implicit default actions like writing to file etc. is not something one would like to have - developer perspective. if she/he wants to serialize the paused object, she can do it by explicitly calling the function. it's there

Reply:
Use case: user runs algorithm and pauses it. Then she want to serialize the intermediate state AND continue training afterwards. Not possible via explicitly calling serializing as that requires training to be stopped fully, and thus it cannot be continued afterwards. So serialization during a pause has to be an option for the user (in addition to do nothing, continue)

Reply:
Furthermore, we dont want to interface with the user to ask for filenames and formats in the pause. This is why there needs to be a way to specify potential intermediate training serialization format and name for the user before training starts

Reply:
why wouldn't you be able to serialize a paused object/method?

Reply:
you can, but ,as said above, you have to stop training for that

Reply:
BTW @shubham808 the file should not be automatically serialized on pausing. Just if the user decides to.

Reply:
```
m = new Machine();

thread 1: m.train();

thread 2: SIG to thread1 pause
thread 2 calls m.save_serializable(...)
does whatever

thread 2 sends a callback to m to continue
```

why wouldn't this work?

Reply:
It works. The idea here was more that, in a typical use case, a user is just living in one thread. So the idea was to make his life easier and offer him a button to press that dumps the intermediate machine somewhere. It is a smallish detail

Reply:
smallish detail but this adds things to the places and functionality places where it does not belong. atm i really dont see this as a good design to go ahead with this as is.